app/tests/test_search.py

```diff
@@ -1755,3 +1922,74 @@ def test_search_transferring_body_returns_correct_date_format(
         assert "01/01/2025" in response.text
 
         assert response.status_code == 200
+
+    @patch("app.main.routes.OpenSearch")
+    def test_search_transferring_body_all_accordions_open_with_open_all_query(
+        self,
+        mock_search_client,
+        client,
+        mock_standard_user,
+        browse_consignment_files,
+    ):
+        """
+        Given a standard user
+        When they make a GET request to the search page with the open_all param
+        All details elements should have an open attribute (which would mean all accordions are open)
+        """
+        mock_search_client.return_value = MockOpenSearch(
+            search_return_value=os_mock_return_tb
+        )
+        mock_standard_user(client, "first_body")
+
+        transferring_body_id = browse_consignment_files[
+            0
+        ].consignment.series.body.BodyId
+
+        form_data = {"query": "test"}
+        response = client.get(
+            f"{self.route_url}/{transferring_body_id}?open_all=open_all",
+            data=form_data,
+        )
+
+        assert response.status_code == 200
+        soup = BeautifulSoup(response.data, "html.parser")
+        checkbox = soup.find("input", {"name": "open_all"})
+        details_elements = soup.find_all("details")
+
+        assert "checked" in checkbox.attrs
+        assert all("open" in details.attrs for details in details_elements)
+
+    @patch("app.main.routes.OpenSearch")
+    def test_search_transferring_body_mark_elements_have_correct_text(
+        self,
+        mock_search_client,
+        client,
+        mock_standard_user,
+        browse_consignment_files,
+    ):
+        """
+        Given a standard user
+        When they make a GET request to the transferring body search page
+        All mark elements should have the correct text content
+        """
+        mock_search_client.return_value = MockOpenSearch(
+            search_return_value=os_mock_return_tb
+        )
+        mock_standard_user(client, "first_body")
+
+        transferring_body_id = browse_consignment_files[
+            0
+        ].consignment.series.body.BodyId
+
+        form_data = {"query": "test"}
+        response = client.get(
+            f"{self.route_url}/{transferring_body_id}", data=form_data
+        )
+
+        assert response.status_code == 200
+        soup = BeautifulSoup(response.data, "html.parser")
+
+        mark_elements = soup.find_all("mark")
```

Comment:
ah I see, you have this test here. Could we not just have this in one test as essentially we just care that given a search with some query, we get results with matches where we have highlights corresponding to the query

Reply:
@anthonyhashemi could merge the two together, the test with raw text also tests that we concat strings together with ellipses " ... "